File: geoutils/test/cgeoutilstest14.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""CGeoUtils test."""
import math
import CGeoUtils as cg
import AltjiraGeoUtils as ag

print "Testing transform graph..."
print "  Creating nodes..."

n0 = ag.LoadSVG.create()
n0.setFileName('test/images/leaf01.svg')
n0.addElementID('leaf01')

n1 = cg.Center.create()
n1.setMethod(cg.CENTER_BOUNDS)

n2 = cg.Normalize.create()

n3 = cg.Scale.create()
n3.setScale(cg.Vector3(600., -600., 1.))

n4 = cg.Rotate.create()
n4.setAngle(0.25 * math.pi)

n5 = cg.WriteSVG.create()
n5.setFileName('temp/nodes02.svg')
n5.setElementIDPrefix('tleaf')
n5.setAxis(cg.AXIS_Z)
n5.setShapeType(cg.SHAPE_TYPE_PATH)

nodes = [ n0, n1, n2, n3, n4, n5 ]
v0 = cg.TransformNodeVector()
i = 0
for it in nodes:
    it.setID("n%02d" % i)
    v0.push_back(it)
    i += 1

print "  Creating graph..."
g0 = cg.TransformGraph()
g0.addNodes(v0)
g0.chain()
g0.setIO()
g0.update()

print "    " + str(g0)
print "    Nodes:"
for i in xrange(0, g0.getNumNodes()):
    print "      " + str(g0.getNode(i))
print "    Connections:"
for i in xrange(0, g0.getNumConnections()):
    print "      " + str(g0.getConnection(i))

print "All done!"

File: geoutils/test/cgeoutilstest14.cpp
#include <cmath>
#include <sstream>
#include <iomanip>
#include "geoutils/geoutils.hpp"
#include "geoutils/altjirageoutils.hpp"

using namespace std;
using namespace Ionflux::GeoUtils;
using namespace Ionflux::GeoUtils::TransformNodes;

int main(int argc, char** argv)
{
    cout << "Testing transform graph..." << endl;
    cout << "  Creating nodes..." << endl;
    
    LoadSVG* n0 = LoadSVG::create();
    n0->setFileName("test/images/leaf01.svg");
    n0->addElementID("leaf01");
    
    Center* n1 = Center::create();
    n1->setMethod(CENTER_BOUNDS);
    
    Normalize* n2 = Normalize::create();
    
    Scale* n3 = Scale::create();
    n3->setScale(Vector3(600., -600., 1.));
    
    Rotate* n4 = Rotate::create();
    n4->setAngle(0.25 * M_PI);
    
    WriteSVG* n5 = WriteSVG::create();
    n5->setFileName("temp/nodes02x.svg");
    
    TransformNodeVector v0;
    v0.push_back(n0);
    v0.push_back(n1);
    v0.push_back(n2);
    v0.push_back(n3);
    v0.push_back(n4);
    v0.push_back(n5);
    
    ostringstream s0;
    unsigned int i = 0;
    for (TransformNodeVector::iterator it = v0.begin(); 
        it != v0.end(); it++)
    {
        s0.str("");
        s0 << "n" << setfill('0') << setw(2);
        (*it)->setID(s0.str());
        i++;
    }
    
    cout << "  Creating graph..." << endl;
    
    TransformGraph g0;
    g0.addNodes(v0);
    g0.chain();
    g0.setIO();
    g0.update();
    
    cout << "    " << g0 << endl;
    cout << "    Nodes:" << endl;
    for (unsigned int i = 0; i < g0.getNumNodes(); i++)
        cout << "      " << *g0.getNode(i) << endl;
    cout << "    Connections:" << endl;
    for (unsigned int i = 0; i < g0.getNumNodes(); i++)
        cout << "      " << *g0.getNode(i) << endl;
    
    cout << "All done!" << endl;
    
    return 0;
}
